### DISCUSSION TITLE
Mapping top level of JSON to a model dict

### DISCUSSION THREAD

**Discussion by bioinformed** [pydantic/pydantic#8971] [Question]
Apologies if this is well-understood, but I'm trying to serialize and deserialize the following JSON structure:

```json
{
    "a": 1,
    "b": 2,
    "c": 3
}
```
into a model like

```python
class MyModel(BaseModel):
    contents = dict[str, int]

    @model_serializer
    def ser_model(self) -> dict[str, int]:
        return self.contents
```
So the model serialization doesn't write a top-level mapping from `contents` to the top-level JSON mapping.  But how do I perform the equivalent transformation upon deserialization?

Thanks in advance...

**Comment by uriyyo** [comment]
Hi @bioinformed,

I guess `RootModel` is what you are looking for - https://docs.pydantic.dev/latest/concepts/models/#rootmodel-and-custom-root-types

```py
from pydantic import RootModel


class MyModel(RootModel[dict[str, int]]):
    pass


m = MyModel.model_validate(
    {
        "a": 1,
        "b": 2,
        "c": 3
    }
)

print(m)
print(m.root)
print(m.model_dump_json(indent=4))
```
```py
root={'a': 1, 'b': 2, 'c': 3}
{'a': 1, 'b': 2, 'c': 3}
{
    "a": 1,
    "b": 2,
    "c": 3
}
```
